pkg/generator/types.go

```diff
@@ -185,9 +186,65 @@ func varNotes(variable broker.BrokerVariable) string {
 		out += fmt.Sprintf(" Default: `%v`.", variable.Default)
 	}
 
+	bullets := constraintsToDoc(variable.ToSchema())
+	if len(bullets) > 0 {
+		out += "\n    * "
+		out += strings.Join(bullets, "\n    * ")
+	}
+
 	return out
 }
 
+// constraintsToDoc converts a map of JSON Schema validation key/values to human-readable bullet points.
+func constraintsToDoc(schema map[string]interface{}) []string {
+	// We use an anonymous struct rather than a map to get a strict ordering of
+	// constraints so they are generated consistently in documentation.
+	// Not all JSON Schema constraints can be cleanly expressed in this format,
+	// nor do we use them all so some are missing.
+	constraintFormatters := []struct {
```

Comment:
This list of formatters seems like it only needs to be defined once. Consider defining as a const outside of the function, perhaps defined before the function ease of reference.

Reply:
I pulled it into a var and a struct, but can't do a const because I guess the structs aren't a `constexpr`.

Reply:
Bummer, in that case keeping it encapsulated inside the function is better from a defensive coding standpoint.

Reply:
moved it back